### PR TITLE
Add work queue problem

### PR DESCRIPTION
**Pull request by mrdooz**
The problem is to expand upon the given simple work queue implementation.

### CODE REVIEW

File: workqueue/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Distributed Work Queue
+For this problem, we want you to build a simple distributed work queue. The system will consist of a client who enqueues some amount of work, and a number of workers who will process the work in parallel, and save the result.
+
+# The Problem
+We have provided a stub implementation, that enqueues work in Redis, spawns a number of worker greenlets who consume the work, and then write the result back to Redis. Your job is to take the our basic implementation, and improve upon it. While this is a very open-ended task, here is a short list of things to consider:
```
Comment (nlal) [inline workqueue/README.md]: take our* basic implementation

File: workqueue/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+* What store/data structure is being used to enqueue unprocessed work? Can you guarantee that each item is only processed by a single worker? If not, is this a problem, and if it is, how do you solve it?
+* If you wanted to ensure that the same work item was only enqueue once, how would you do that?
+* The currently implementation doesn't support any notion of priorities. If you wanted to add this, how could that be done?
```
Comment (nlal) [inline workqueue/README.md]: The current*

File: workqueue/workqueue_runner.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
+"""Distributed work queue test harness."""
+
+# Enqueues work items in Redis, waits N seconds, and then tries to validate
```
Comment (nlal) [inline workqueue/workqueue_runner.py]: i think these comments should be part of the docstring

Comment (nlal) [inline workqueue/workqueue_runner.py]: oh i see why you did this

Comment (mrdooz) [inline workqueue/workqueue_runner.py]: Yeah, I went back and forth on that.. When they were part of it, argparse --help spat out the whole thing in a single unformatted block, but I agree that in the code it looks better as a single docstring.

**Comment by nlal** [comment]
Cool, ran it locally and it worked with 1 worker and failed with 2. LGTM.

**Pull request closed without merging**